build.gradle

```
@@ -9,18 +9,19 @@ repositories {
 
 dependencies {
     // main dependencies
-    implementation group: 'org.codehaus.groovy', name: 'groovy-all', version: '2.4.12'
-    implementation group: 'com.cloudbees', name: 'groovy-cps', version: '1.32'
-    implementation group: 'org.jenkins-ci.main', name: 'jenkins-core', version: '2.277'
+    implementation group: 'org.codehaus.groovy', name: 'groovy-all', version: '2.4.21'
+    implementation group: 'com.cloudbees', name: 'groovy-cps', version: '3785.vee73da_b_9544e'
+    implementation group: 'org.jenkins-ci.main', name: 'jenkins-core', version: '2.414.1'
+    implementation group: 'com.google.guava', name: 'guava', version: '32.1.2-jre'
```

I'd add a comment to explain this one, something like:
```suggestion
    // jenkins-core has a transitive dependency on guava 32.1.1-jre, but that version won't load for gradle 
    // when jenkins-core is upgraded past 2.414.1 this dependency should be re-evaluated and hopefully removed
    implementation group: 'com.google.guava', name: 'guava', version: '32.1.2-jre'
```